examples/SmartBank/test/SmartBankTest.js

```diff
@@ -0,0 +1,171 @@
+  return decodedResult.value
+}
+function decodeContractAddress (key) {
+  const decoded58address = Crypto.decodeBase58Check(key.split('_')[1]).toString(
+    'hex'
+  )
+  return `0x${decoded58address}`
+}
+
+describe('SmartBank Contract', () => {
+  let owner
+  let SmartBankContract
+
+  before(async () => {
+    const ownerKeyPair = wallets[0]
+    owner = await Ae({
+      url: config.host,
+      internalUrl: config.internalHost,
+      keypair: ownerKeyPair,
+      nativeMode: true,
+      networkId: 'ae_devnet'
+    })
+  })
+
+  describe('Deploy contract', () => {
+    it('should deploy SmartBank contract', async () => {
+      const gas = { gas: config.gas }
+      const deployObj = {
+        options: { ttl: config.ttl }
+      }
+      SmartBankContract = await deployContract(
+        'SmartBank',
+        owner,
+        gas,
+        deployObj
+      )
+      assert(SmartBankContract.hasOwnProperty('address'))
+      assert(SmartBankContract.hasOwnProperty('owner'))
+    })
+  })
+
+  describe('Interact with the contract', () => {
+
+    it('should deposit tokens', async () => {
+      const args = {
+        args : '("12345")',
+        options: { ttl: 55 , amount: 1000},
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        SmartBankContract,
+        'deposit',
+        args,
+        '()'
+      )
+      assert.deepEqual(result, [])
+
+    })
+    it('should check if user is registered in state', async () => {
+      const key = decodeContractAddress(wallets[0].publicKey)
+      const args = {
+        args : key, // pass the address of the caller
+        options: {ttl: 55},
+        abi: 'sophia'
+      }
+      wallets[0].publicKey
+      const result = await callContract(
+        SmartBankContract,
+        'user_exists',
+        args,
+        'bool'
+      )
+      assert.equal(result, '1')
+    })
+
+
+    it('should transfer tokens to its owner (withdrawal)', async () => {
+      const args = {
+        args: `("12345", 500)`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        SmartBankContract,
+        'withdraw',
+        args,
+        '()'
+         // return type?
+      )
+      assert.deepEqual(result, [])
+    })
+
+    it('should check new balance after withdrawal', async () => {
+      const args = {
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        SmartBankContract,
+        'check_self_balance',
+        args,
+        )
+      assert.equal(result, 500)
+    })
+
+    it('should transfer tokens', async () => {
+      const key = decodeContractAddress(wallets[0].publicKey)
+      const args = {
+        args: `(${key}, 250, "12345")`,
+        options: { ttl: 55 },
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        SmartBankContract,
+        'transfer',
+        args)
+      assert.equal(result,  [])
```

Comment:
are you transfer tokens from address 'AAA' to 'AAA', same address ? test to transfer from address 'AAA' to 'BBB'

Reply:
To make this happen , first of all, the 'BBB' account should deposit some tokens, in order to be in contract's state, and only then we can perform a transfer from 'AAA' (which also should be in the state and have some tokens) to 'BBB'. The problem is, I can't really understand how to call a contract from another account. As far as I understood, first of all I should initialize another 'AE Universal' instance with another keypair loaded in, but what next?  How to perform a call to a contract that is already deployed from another keypair? What parameters and which function should be called from another instantiated client?